tests/Pyxsim/pyxsim.py

```diff
@@ -0,0 +1,466 @@
+        c_package = c_char_p(package)
+        c_pin = c_char_p(pin)
+        c_value = c_int()
+        status = xsi_lib.xsi_is_pin_driving(self.xsim, c_package, c_pin, byref(c_value))
+        XsiStatus.error_if_not_valid(status)
+        return c_value.value
+
+    def is_port_pins_driving(self, tile, port):
+        c_tile = c_char_p(tile)
+        c_port = c_char_p(port)
+        c_value = c_int()
+        status = xsi_lib.xsi_is_port_pins_driving(
+            self.xsim, c_tile, c_port, byref(c_value)
+        )
+        XsiStatus.error_if_not_valid(status)
+        return c_value.value
+
+    def read_mem(self, tile, address, num_bytes, return_ctype=False):
+        c_tile = c_char_p(tile)
+        c_address = c_int(address)
+        c_num_bytes = c_int(num_bytes)
+        buf = create_string_buffer(num_bytes)
+        status = xsi_lib.xsi_read_mem(self.xsim, c_tile, c_address, c_num_bytes, buf)
+        XsiStatus.error_if_not_valid(status)
+        if return_ctype:
+            return buf
+        else:
+            return list(buf)
+
+    def read_symbol_word(self, tile, symbol, offset=0):
+        address = self.xe.symtab[tile, symbol]
+        address += offset
+        buf = self.read_mem(tile, address, 4, return_ctype=True)
+        return struct.unpack("<I", buf)
+
+    def read_symbol_byte(self, tile, symbol, offset=0):
+        address = self.xe.symtab[tile, symbol]
+        address += offset
+        buf = self.read_mem(tile, address, 1, return_ctype=True)
+        return ord(buf[0])
+
+    def write_mem(self, tile, address, num_bytes, data):
+        c_tile = c_char_p(tile)
+        c_address = c_int(address)
+        c_num_bytes = c_int(num_bytes)
+        buf = create_string_buffer(data)
+        status = xsi_lib.xsi_write_mem(self.xsim, c_tile, c_address, c_num_bytes, buf)
+        XsiStatus.error_if_not_valid(status)
+
+    def write_symbol_word(self, tile, symbol, value, offset=0):
+        address = self.xe.symtab[tile, symbol]
+        address += offset
+        data = struct.pack("<I", value)
+        self.write_mem(tile, address, 4, data)
+
+    def write_symbol_byte(self, tile, symbol, value, offset=0):
+        address = self.xe.symtab[tile, symbol]
+        address += offset
+        data = struct.pack("<c", value)
+        self.write_mem(tile, address, 1, data)
+
+    def read_pswitch_reg(self, tile, reg_num):
+        c_tile = c_char_p(tile)
+        c_reg_num = c_int(reg_num)
+        c_value = c_int()
+        status = xsi_lib.xsi_read_pswitch_reg(
+            self.xsim, c_tile, c_reg_num, byref(c_value)
+        )
+        XsiStatus.error_if_not_valid(status)
+        return c_value.value
+
+    def write_pswitch_reg(self, tile, reg_num, value):
+        c_tile = c_char_p(tile)
+        c_reg_num = c_int(reg_num)
+        c_value = c_int(value)
+        status = xsi_lib.xsi_write_pswitch_reg(self.xsim, c_tile, c_reg_num, c_value)
+        XsiStatus.error_if_not_valid(status)
+
+
+class XsiPlugin(object):
```

Comment:
Could this class inherit from ```abc.ABC``` instead of ```object```? Looks like it's only function is to enforce the implementation of a function called ```clock``` to me...

Reply:
Can you raise an issue for this one?